src/abstraction/transient.rs

```diff
@@ -363,22 +372,23 @@ impl TransientKeyContextBuilder {
     /// * it is the responsibility of the client to ensure that the context can be initialized
     /// safely, threading-wise
     /// * the client is also responsible of choosing the correct TCTI to connect to.
+    /// * it is the responsability of the client to set a sufficiently secure default cipher for the context
     ///
     /// # Constraints
-    /// * `root_key_size` must be 1024 or 2048
+    /// * `root_key_size` must be 1024, 2048 or 4096
     /// * `root_key_auth_size` must be at most 32
     ///
     /// # Errors
     /// * errors are returned if any method calls return an error: `Context::get_random`,
     /// `Context::start_auth_session`, `Context::create_primary_key`, `Context::flush_context`,
     /// `Context::set_handle_auth`
     /// * if the root key authentication size is given greater than 32 or if the root key size is
-    /// not 1024 or 2048, a `WrongParamSize` wrapper error is returned
+    /// not 1024, 2048 or 4096, a `WrongParamSize` wrapper error is returned
     pub unsafe fn build(self) -> Result<TransientKeyContext> {
         if self.root_key_auth_size > 32 {
             return Err(Error::local_error(ErrorKind::WrongParamSize));
         }
-        if self.root_key_size != 1024 && self.root_key_size != 2048 {
+        if self.root_key_size != 1024 && self.root_key_size != 2048 && self.root_key_size != 4096 {
             error!("The reference implementation only supports key sizes of 1,024 and 2,048 bits.");
```

Comment:
I think we first set it this way because of the message in the `error` below, that the reference implementation is only supporting key sizes of 1024 and 2048 bits and that we were targeting that reference implementation. Is that fine to add 4096 here?

Reply:
"The reference implementation" is only really a simulator, ideally we'd be targeting physical implementations. Tbh implementations could support any size, not only powers of two, and now that we have `TestParms` we can just relegate that check to users...

Reply:
We might have had that discussion before but in my mind "implementation" in that context means the ESYS API implementation, ie the library that we link to in this crate.

Reply:
As in, even before the commands each the physical TPM, the calls might fail in the TSS because this size is not supported. At least that's what I understood from this sentence.

Reply:
ah, true, I forgot about that; will check again just to make sure

Reply:
Interestingly, this time the specs took me in a different direction.

So `TPMI_RSA_KEY_BITS` is supposed to be the maximum size for a specific RSA key, and implementations should allow any value that is one of `RSA_KEY_SIZES_BITS` which are the predefined allowed values for RSA key sizes. This constant is defined [here](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part4_SuppRoutines_code_pub.pdf) on page 634 and allows RSA keys to be of 1024, 2048, 3072 or 4096 bits. So, if anything, we're actually missing one value. I think the TSS might default to a maximum of 2048 but might allow for up to 4096 if compiled with certain params?

Reply:
oh good to know! I guess it is fine to support that then.

Reply:
I've updated this to cover 3072 bits as well and changed the way the check is done